workflow/Snakefile

```diff
@@ -74,6 +84,57 @@ rule create_results_directory:
         touch {output.TEMP_}
         """
 
+rule plot_sequence_logos:
+    """
+    Create directories for the results
+    """
+    input:
+        FILE_pwm = expand(
+                os.path.abspath(
+                os.path.join(
+                    config["pwm_directory"],
+                    "{pwm_dir}"
+                    )
+                ),
+                pwm_dir = gather_motifs_names(config["pwm_directory"])
+            ),
+        SCRIPT_ = os.path.join(
+            config["pipeline_path"],
+            "workflow",
+            "scripts",
+            "sequence_logos.py"
+        )
+            
+    output:
+        PNG_sequence_logos = expand( 
+            os.path.join(
+                "{outdir}",
+                "sequence_logos",
+                "{pwm_dir}.png"
+            ),
+            outdir = config["outdir"],
+            pwm_dir = gather_motifs_names(config["pwm_directory"])
+            )
+
+    params:
+        DIR_sequence_logos = expand( 
```

Comment:
`expand` is to be used under `input` of the following rule, not under `output` of the rule that is being parallelized. Think of it that there are 2 rules: former and latter. The latter executes the expansion and asks for many files. The former catches each of these request by providing output files in the form of wildcards.

Reply:
Currently, the output of the rule expects the `png` of all the pwm motifs, if not `expand`, how should we write it?

Reply:
This comment applies to both the expansions in the `params` and `output`.
Wildcards will flow from the subsequent rule (which in this case would be `all` rule) so, as I said, you do not use the keyword `expand` here. You may just specify the variables under output/params with pure wildcard variables: `variable = "{wildcard}"`

Please read on `expand` in snakemake.